### PR TITLE
chore: bump matrix_sdk lib

### PR DESCRIPTION
**Pull request by guidota**
## Description

The last release of `matrix_sdk` was last year and I found a few deadlocks when registering and loging users.

There are a lot of fixes right now since then in the main branch. I think we can stick to a specific revision of this library and see how it goes.

### TO DO

- [x] Better homeserver URL handling.
- [x] Check if we can get more information from http errors.

### CODE REVIEW

File: src/user.rs [hunk 1]
```diff
@@ -297,28 +301,110 @@ impl User<Synching> {
...
-        match response {
-            Ok(response) => {
-                self.send(Event::RequestDuration((
-                    UserRequest::SendMessage,
-                    instant.elapsed(),
-                )))
-                .await;
-
-                self.send(Event::MessageSent(response.event_id.to_string()))
+        if let Some(room) = client.get_joined_room(room_id) {
```
Comment (moliva) [inline src/user.rs:308]: I would `unwrap_or_else` here, the join **must** be there already.

File: src/user.rs [hunk 2]
```diff
@@ -297,28 +301,110 @@ impl User<Synching> {
...
+        let parts: Vec<&str> = regex.splitn(homeserver, 2).collect();
+        (parts[1], homeserver.to_string())
+    } else {
+        let protocol = protocol.unwrap_or("https");
+        (homeserver, format!("{protocol}://{homeserver}"))
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (moliva) [inline src/user.rs:368]: 👏🏻 ✨

File: src/user.rs [hunk 3]
```diff
@@ -297,28 +301,110 @@ impl User<Synching> {
...
+            "User {} received a message from room {} and sent by {}",
+            user_id,
+            room.room_id(),
+            event.sender
+        );
+    }
+}
+
+// /// This function returns homeserver domain and url, ex:
+// ///  - get_homeserver_url("matrix.domain.com") => ("matrix.domain.com", "https://matrix.domain.com")
```
Comment (moliva) [inline src/user.rs:355]: Let's leave the comments as actual documentation for the function.
```suggestion
/// This function returns homeserver domain and url, ex:
///  - get_homeserver_url("matrix.domain.com") => ("matrix.domain.com", "https://matrix.domain.com")
```

**Review by moliva — APPROVED** [review]
Looks good and works. Please see comments before merging.

File: src/user.rs (hunk 1 above)
Comment (guidota) [reply src/user.rs:308]: That's true. I'd go for `expect` since we don't have a default room to provide to `unwrap_or_else`. Wdyt?

**Pull request merged**